docs/config/shared-options.md

```diff
@@ -258,6 +258,31 @@ export default defineConfig({
 })
 ```
 
+You can pass a function to specific different option for different files. Note it only works for file imported from javascript, or in other word —— the entry style file.
```

Comment:
Doesn't preprocess apply to all CSS files regardless of how it's imported? E.g. `@import` styles are preprocessed at https://github.com/vitejs/vite/blob/3aab14eb25446c0c5830a504b34d39ce434e37d6/packages/vite/src/node/plugins/css.ts#L1037-L1047

Reply:
I tried in playground . If a less file is only imported by other less file ,  the preprocessoroptions will not regenerate for the file. Only the entry less file get a new option.

Reply:
Looks like less or scss compiler deal with all imported files with one option from the entry one.